src/main/java/uk/gov/hmcts/reform/civil/handler/callback/migration/MigrateCaseDataCallbackHandler.java

```diff
@@ -31,21 +35,72 @@ public class MigrateCaseDataCallbackHandler extends CallbackHandler {
 
     private static final List<CaseEvent> EVENTS = Collections.singletonList(migrateCase);
 
-    private static final String MIGRATION_ID_VALUE = "AccessProfileMigration";
+    private static final String MIGRATION_ID_VALUE = "GSMigration";
 
     private final ObjectMapper objectMapper;
 
+    private final CoreCaseDataService coreCaseDataService;
+    private final LocationRefDataService locationRefDataService;
+
     @Override
     protected Map<String, Callback> callbacks() {
         return new ImmutableMap.Builder<String, Callback>()
             .put(callbackKey(ABOUT_TO_SUBMIT), this::migrateCaseData)
+            .put(callbackKey(SUBMITTED), this::migrateSuppmentryData)
             .build();
     }
 
     private CallbackResponse migrateCaseData(CallbackParams callbackParams) {
         CaseData oldCaseData = callbackParams.getCaseData();
         CaseData.CaseDataBuilder<?, ?> caseDataBuilder = oldCaseData.toBuilder();
 
+        log.info("Migrating data for case: {}", oldCaseData.getCcdCaseReference());
+        String authToken = callbackParams.getParams().get(BEARER_TOKEN).toString();
+
+        CaseLocation caseLocation = CaseLocation.builder().baseLocation("420219").region("2").build();
+        if (CaseCategory.SPEC_CLAIM.equals(oldCaseData.getCaseAccessCategory())) {
```

Comment:
Make sure that the systemUpdateUser has access to these fields in order to read/update case data